environment_unix.yml

```diff
@@ -8,6 +8,7 @@ dependencies:
   - cmake
   - gtest
   - gflags
+  - benchmark
```

Comment:
Is this really necessary here? Given that we're bundling it as a submodule

Reply:
I tried building in conda without it and it failed. (following the instructions in the contributing docs).

Command `cmake -DTEST=off --preset linux-conda-debug cpp` failed with
```
CMake Error at arcticdb/CMakeLists.txt:82 (find_package):
  By not providing "Findbenchmark.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "benchmark", but CMake did not find one.

  Could not find a package configuration file provided by "benchmark" with
  any of the following names:

    benchmarkConfig.cmake
    benchmark-config.cmake

  Add the installation prefix of "benchmark" to CMAKE_PREFIX_PATH or set
  "benchmark_DIR" to a directory containing one of the above files.  If
  "benchmark" provides a separate development package or SDK, be sure it has
  been installed.
```

Let me know if you have suggestions on how to fix it.

Also it looks like there are other submodules which we include in the .yml (e.g. pybind11, rapidcheck etc.). If we can remove this it's probably better suited for a separate PR.

Reply:
OK yes I think I'm misunderstanding the Conda build system here. Fine to leave this as is.